### PR TITLE
Refactor UI, Update Theme, and Add Tests for Sign In and Profile Creation

### PR DESCRIPTION
**Pull request by AlexisAllmnd**
# Description

This pull request introduces several important changes to the app, including significant UI improvements, theme updates, and the addition of initial tests for the sign-in and profile creation processes. These changes improve the user experience and ensure a more cohesive design across the app.

### 1. **UI Refactor for Profile and Schedule Creation**
   - **ProfileUserCreation.kt**: Redesigned the user profile creation screen with improved layout, typography, and button styles. Added better field validation and error handling.
   - **ProfileTutorCreation.kt**: Enhanced the tutor profile creation screen with better language and subject selection UI. Updated the slider for selecting tutoring prices with modernized styling.
   - **ProfileTutorSchedule.kt**: Improved the tutor schedule creation screen, ensuring time slots and hour labels have consistent heights. Adjusted the grid layout to make availability selection clearer and easier to use.

### 2. **Theme Updates**
   - Refined the color palette for both light and dark modes to align with the new visual style of the app.
   - Updated `MainActivity.kt` to integrate the new color schemes and layouts.
   - Adjusted typography for titles, body text, and headings to improve readability.
   - Updated theme-related files (`Color.kt`, `Theme.kt`, `Typography.kt`) to ensure consistency throughout the app.

### 3. **Main Activity Adjustments**
   - Updated `MainActivity.kt` to reflect the renaming and restructuring of profile creation functions (e.g., renamed `TutorInfoScreen` to `CreateTutorProfile`).
   - Applied color scheme changes and adapted the navigation actions to reflect the new flow for tutor profiles and schedules.

### 4. **Additional Subjects for Tutors**
   - Expanded the list of subjects tutors can select during profile creation, including **AICC, ICC, and Coding** as new subjects.

### 5. **Initial Test Cases**
   - **SignInTest.kt**: Updated test for verifying UI elements such as the logo, button, and image scroll functionality. Added checks for all images in the onboarding carousel.
   - **CreateUserProfileTest.kt**: Added new test cases for the profile creation screen, including checks for:
     - Rendering correctness of key elements (e.g., text fields, role selection buttons).
     - Form validation when required fields are empty.
     - Phone number validation to ensure correct input format.
     - Role selection functionality to ensure that the selected role is correctly updated.
   - These tests are foundational and require further development, but they provide a good starting point for ensuring UI correctness.

### 6. **Minor Bug Fixes**
   - Corrected small layout issues, including inconsistent padding and spacing.
   - Fixed minor issues with the validation logic in the profile creation process.

## Motivation

The previous UI and theme were outdated and lacked consistency. These changes bring a modernized look and feel to the app, aligning it with current design trends. The tests are part of the ongoing effort to increase code coverage and ensure the reliability of the app.

## Testing

- **Manual Testing**: Conducted on different screen sizes and devices to ensure responsiveness and UI consistency.
- **Automated Testing**: Added new tests for sign-in and profile creation using `composeTestRule`.
- **Validation**: All test cases passed on local builds, but additional tests will be developed to cover more edge cases.

## Next Steps

- Complete the remaining test cases for the profile and sign-in flows.
- Refactor backend logic to better integrate with the new UI, focusing on profile data storage and retrieval.
- Add transitions and animations to enhance the user experience during navigation.

# Affected Files

- `ProfileUserCreation.kt`
- `ProfileTutorCreation.kt`
- `ProfileTutorSchedule.kt`
- `MainActivity.kt`
- `SignInTest.kt`
- `CreateUserProfileTest.kt`
- `CreateTutorSchedule.kt`
- `theme.xml`, `Color.kt`, `Theme.kt`, `Typography.kt`

### CODE REVIEW

File: app/src/main/java/com/github/se/project/ui/profile/CreateTutorSchedule.kt [hunk 1]
```diff
@@ -1,40 +1,13 @@
-package com.github.se.project.ui.profile
```
Comment (aferrante1) [inline app/src/main/java/com/github/se/project/ui/profile/CreateTutorSchedule.kt:1]: did you remove the package completely with out replacing it?

File: app/src/main/java/com/github/se/project/MainActivity.kt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package com.github.se.project
 
+import CreateTutorSchedule
```
Comment (aferrante1) [inline app/src/main/java/com/github/se/project/MainActivity.kt:3]: the import should be sometihnig like import com.github.se.project.ui.profile.CreateTutorSchedule

**Comment by aferrante1** [comment]
The UI looks great and easy to use good job on that!

Just a comment unrelated to your changes, when I signIn as a Student, everytime I rebuild the app it logs me to the HomeScreen whereas if I do the same with the Tutor I let me create a new account everytime.

File: app/src/main/java/com/github/se/project/ui/profile/CreateTutorSchedule.kt [hunk 3]
```diff
@@ -58,83 +31,72 @@ fun AvailabilityScreen(
...
-                  text =
-                      "Finish your account creation by selecting the time slots you're available during the week:",
-                  fontSize = 16.sp,
-                  modifier = Modifier.padding(bottom = 8.dp))
+                  "Finish your account creation by selecting the time slots you're available during the week:",
+                  style = MaterialTheme.typography.bodyLarge,
+                  modifier = Modifier.testTag("instructionText"))
 
-              // Display the availability grid
               AvailabilityGrid(
```
Comment (aferrante1) [inline app/src/main/java/com/github/se/project/ui/profile/CreateTutorSchedule.kt:53]: The 19h row looks off and is overlapping the 18h row when testing on my phone
IMG_3014

**Review by aferrante1 — COMMENTED** [review]
Looks great, good job!

**Review by aferrante1 — APPROVED** [review]
